#!/usr/bin/env python
PACKAGE="riptide_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

#        Name               Type     Level  Description                       Default   Min       Max
gen.add( "Mass" ,           double_t, 1,    "Mass.",                          35.7,      0,       100)
gen.add( "Volume" ,         double_t, 1,    "Volume.",                        0.03575,   0,       0.2)
gen.add( "Buoyancy_X_POS" , double_t, 1,    "Buoyancy X Pos.",                0,        -0.5,     0.5)
gen.add( "Buoyancy_Y_POS" , double_t, 1,    "Buoyancy Y Pos.",                0,        -0.5,     0.5)
gen.add( "Buoyancy_Z_POS" , double_t, 1,    "Buoyancy Z Pos.",                0,        -0.5,     0.5)
gen.add( "SSL_active" ,     bool_t,   1,    "SSL active?",                    True)
gen.add( "SPL_active" ,     bool_t,   1,    "SPL active?",                    True)
gen.add( "HSF_active" ,     bool_t,   1,    "HSF active?",                    True)
gen.add( "HPF_active" ,     bool_t,   1,    "HPF active?",                    True)
gen.add( "HSA_active" ,     bool_t,   1,    "HSA active?",                    True)
gen.add( "HPA_active" ,     bool_t,   1,    "HPA active?",                    True)
gen.add( "SWF_active" ,     bool_t,   1,    "SWF active?",                    True)
gen.add( "SWA_active" ,     bool_t,   1,    "SWA active?",                    True)
                 # PkgName  #NodeName         #Prefix for generated .h include file, e.g. ParametersConfig.py
exit(gen.generate(PACKAGE, "thruster_controller", "VehicleProperties"))
